### PR TITLE
[Wallet] Fix invite redemption gas handling and gas price calculation

### PR DESCRIPTION
**Pull request by jmrossy**
## Description

Calculate gas for withdrawing from invite codes more precisely

## Other changes

Improve organization of some gas related things

## Tested

As part of alfajroes release prep

## Backwards compatibility

Yes

## Commit Message
[Wallet] Fix invite redemption gas handling and gas price calculation

The app now calculates invite code withdrawal fees dynamically and also inflates gas prices to match what Geth does.

### CODE REVIEW

File: packages/mobile/src/web3/utils.ts [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+import BigNumber from 'bignumber.js'
+import { GAS_INFLATION_FACTOR } from 'src/config'
 import Logger from 'src/utils/Logger'
 import { getContractKit } from 'src/web3/contracts'
+import { Tx } from 'web3-core'
+import { TransactionObject } from 'web3-eth'
 
 const TAG = 'web3/utils'
 
+// Estimate gas taking into account the cofigured inflation factor
```
Comment (jeanregisser) [inline packages/mobile/src/web3/utils.ts]: nit: typo `configured`

File: packages/mobile/src/invite/saga.ts [hunk 2]
```diff
@@ -353,19 +354,39 @@ export function* withdrawFundsFromTempAccount(
...
+  const sendTokenFee = divideByWei(sendTokenFeeInWei).times(5)
+
+  if (sendTokenFee.isGreaterThanOrEqualTo(tempAccountBalance)) {
+    throw new Error('Fee is too large for amount in temp wallet')
+  }
+
+  const netSendAmount = tempAccountBalance.minus(sendTokenFee)
+  Logger.debug(
+    TAG + '@withdrawFundsFromTempAccount',
+    `Withdrwaing net amount of ${netSendAmount.toString()}`
```
Comment (jeanregisser) [inline packages/mobile/src/invite/saga.ts]: nit: typo `Withdrawing`

File: packages/mobile/src/invite/saga.ts [hunk 3]
```diff
@@ -353,19 +354,39 @@ export function* withdrawFundsFromTempAccount(
...
     recipientAddress: newAccount,
+    amount: tempAccountBalance,
     comment: SENTINEL_INVITE_COMMENT,
-    // TODO: appropriately withdraw the balance instead of using gas fees will be less than 1 cent
-    amount: tempAccountBalance.minus(0.01).toString(),
   })
+  const sendTokenFee = divideByWei(sendTokenFeeInWei).times(5)
+
+  if (sendTokenFee.isGreaterThanOrEqualTo(tempAccountBalance)) {
+    throw new Error('Fee is too large for amount in temp wallet')
```
Comment (jeanregisser) [inline packages/mobile/src/invite/saga.ts:372]: nice! this way we'll see the error more easily :+1:

Wondering, shouldn't we also check that the amount is enough to do verification as well?

**Review by cmcewen — APPROVED** [review]

**Review by jeanregisser — APPROVED** [review]
Much better 🎉 👍 
Left one question.

File: packages/mobile/src/invite/saga.ts (hunk 3 above)
Comment (jmrossy) [reply packages/mobile/src/invite/saga.ts:372]: Good idea! It's a bit tough to know exactly what verification will cost but I suppose we already hard code an estimate in choosing the invite amount. I'll do this in a separate PR

**Pull request merged**